### PR TITLE
fix SET CHARSET | NAMES DEFAULT

### PR DESCRIPTION
**Pull request by sim41**
### What problem does this PR solve? 
https://dev.mysql.com/doc/refman/8.0/en/set-character-set.html
It is similar to the issue: https://github.com/pingcap/parser/issues/498. set charset default syntax, but cause of different syntax.

### MySQL Syntax:
```
SET {CHARACTER SET | CHARSET}
    {'charset_name' | DEFAULT}
```

### Bad SQL Case:
```
 set names default;
 set character set default;
 set charset default;
 set char set default;
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -686,6 +686,7 @@ import (
 	SignedLiteral			"Literal or NumLiteral with sign"
 	DefaultValueExpr		"DefaultValueExpr(Now or Signed Literal)"
 	NowSymOptionFraction		"NowSym with optional fraction part"
+        CharsetNameOrDefault            "Character set name or default"
```
Comment (tangenta) [inline parser.y]: Please reindent this line with tab.

File: parser.y [hunk 2]
```diff
@@ -7326,13 +7327,25 @@ VariableAssignment:
...
+
+CharsetNameOrDefault:
+        CharsetName
+        {
+                $$ = ast.NewValueExpr($1.(string))
+        }
+|       "DEFAULT"
+        {
+                $$ = &ast.DefaultExpr{}
+        }
```
Comment (tangenta) [inline parser.y]: Ditto

File: parser.y (hunk 1 above)
Comment (sim41) [reply parser.y]: sry, I used to set expandtab in vim, and gofmt does not work in *.y file. I'll reindent the tab.

**Comment by leoppro** [comment]
LGTM

**Comment by leoppro** [comment]
@tiancaiamao  PTAL

**Comment by tiancaiamao** [comment]
LGTM
Please resolve conflicts @sim41

**Pull request merged**